overlord/state/taskrunner.go

```diff
@@ -163,13 +159,12 @@ func (r *TaskRunner) Stop() {
 	}
 }
 
-// Settle waits for all concurrent activities and returns after that's done.
-func (r *TaskRunner) Settle() {
+// Wait waits for all concurrent activities and returns after that's done.
+func (r *TaskRunner) Wait() {
```

Comment:
Per conversation last night, this method cannot hold the lock while waiting.

Something like this should work:
```
s.mu.Lock()
defer s.mu.Unlock()
for len(r.tombs) > 0 {
        for id, t := range r.tombs {
                s.mu.Unlock()
                t.Wait()
                s.mu.Lock()
                delete(r.tombs, id)
                break
        }
}
```